timeseries/src/autogluon/timeseries/models/autogluon_tabular/tabular_model.py

```diff
@@ -183,29 +191,21 @@ def extend_single_time_series(group):
         return data.groupby(ITEMID, sort=False).apply(extend_single_time_series)
 
     def predict(self, data: TimeSeriesDataFrame, quantile_levels: List[float] = None, **kwargs) -> TimeSeriesDataFrame:
-        if quantile_levels is not None:
-            output_quantiles = quantile_levels
-            if not set(quantile_levels).issubset(set(self.quantile_levels)):
-                raise ValueError(
-                    f"{self.name} expects quantile_levels to be a subset of {self.quantile_levels} "
-                    f" (received quantile_levels = {quantile_levels})."
-                )
-        else:
-            output_quantiles = self.quantile_levels
-            if self._drop_median_prediction:
-                output_quantiles.remove(0.5)
+        self._check_predict_inputs(data=data, quantile_levels=quantile_levels)
+        if quantile_levels is None:
+            quantile_levels = self.quantile_levels
 
         data, scale_per_item = self._normalize_targets(data)
         data_extended = self._extend_index(data)
         features = self._get_features_dataframe(data_extended, last_k_values=self.prediction_length)
         predictions = self.tabular_predictor.predict(features[self._available_features])
 
+        predictions = predictions.rename("mean").to_frame()
         preds_index = data_extended.slice_by_timestep(-self.prediction_length, None).index
         predictions.set_index(preds_index, inplace=True)
 
-        predictions["mean"] = predictions[0.5]
-        predictions = predictions[["mean"] + output_quantiles]
-        predictions.rename(str, axis=1, inplace=True)
+        for q in quantile_levels:
+            predictions[str(q)] = predictions["mean"] + self.residuals_std * scipy.stats.norm.ppf(q)
```

Comment:
this is very cool! this also means that with very minimal effort, thanks to `scipy` we can generalize the residual distributions, which probably will bring a huge boost to performance.